### DISCUSSION TITLE
Is it possible to suspend component on refetch?

### DISCUSSION THREAD

**Discussion by bpinto** [TanStack/query#7430] [Q&A]
I have two questions that I might have missed when reading the documentation.

Is it possible to cause a component to suspend when a useSuspenseQuery is refetching?

Is it possible to suspend a component on mount if the cached data is stale?

If not, would this be something interesting to have available on the library? I was thinking about an option like: `{ suspendOn: "refetch", "staleFetch", "always", "noData" }`

**Comment by TkDodo** [comment]
> Is it possible to cause a component to suspend when a useSuspenseQuery is refetching?

no

> Is it possible to suspend a component on mount if the cached data is stale?

no

> If not, would this be something interesting to have available on the library?

no, not really, because the premise of `stale-while-revalidate` is that stale data is better than no data / a fallback loading spinner. Making the API surface more complex is not something that we're willing to do lightly for a use-case that we don't think is desireable.

Out of curiosity: Why would you want that?

Additionally, you can always look at `isFetching` and then show a loading spinner if you want, or trigger the suspense boundary manually (if that's even possible, not sure)

**Comment by bpinto** [comment]
> no, not really, because the premise of stale-while-revalidate is that stale data is better than no data / a fallback loading spinner. Making the API surface more complex is not something that we're willing to do lightly for a use-case that we don't think is desireable.

That's fair. 👍 

> Out of curiosity: Why would you want that?

For our app, we do not believe that on every occasion we provide a better user experience by showing stale data and adding a loader to indicate that data is being updated in the background.

There are occasions where we don't want that to happen, e.g. forms (we've read your blog post on this), or when we are sure that the data has been previously updated (e.g. `invalidateQueries` was invoked).

With the introduction of `startTransition()`, IMO this could be handled by react's mechanism instead of using `isFetching` to show or not a spinner.

> Additionally, you can always look at isFetching and then show a loading spinner if you want, or trigger the suspense boundary manually (if that's even possible, not sure)

Thank you 👍

**Comment by bpinto** [comment]
> Additionally, you can always look at isFetching and then show a loading spinner if you want, or trigger the suspense boundary manually (if that's even possible, not sure)

I'm struggling to implement. I think it would be possible if `fetchQuery`'s promise was not recreated on every invocation. I'll keep trying other ideas though.
